src/vmspawn/vmspawn.c

```diff
@@ -685,6 +783,58 @@ static int run_virtual_machine(void) {
         } else
                 log_warning("Cannot append extra args to kernel cmdline, native architecture doesn't support SMBIOS");
 
+        /* disable TPM autodetection if the user's hardware doesn't support it */
+        if (!ARCHITECTURE_SUPPORTS_TPM) {
+                if (arg_tpm < 0) {
+                        arg_tpm = 0;
+                        log_debug("TPM not support on %s, disabling tpm autodetection and continuing", architecture_to_string(native_architecture()));
+                } else if (arg_tpm > 0)
+                        return log_error_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "TPM not supported on %s, aborting", architecture_to_string(native_architecture()));
```

Comment:
so far we avoided the word "aborting" for something that is not abort(), i.e. exits with a signal.

Maybe just drop the ", aborting" alotgether? or replace by "refusing" or so?